audio/drivers/alsa.c

```diff
@@ -197,7 +197,7 @@ static ssize_t alsa_write(void *data, const void *buf_, size_t size_)
 
    if (alsa->nonblock)
    {
-      while (size)
+      while (size > 0)
```

Comment:
This is not meaningful, size cannot become negative due to the write loop. The assert should be that `size >= 0`. The writei call takes size as a maximum number of frames to write, and then size is decremented based on that.

Reply:
OK understood, but then size >=0 will also be wrong because the loop carries on until size reaches zero.  If we change to while (size >= 0) the loop will never end?  Have I misunderstood your point?

In any case, if we can guarantee size will never be negative then agree that my change is not necessary to the current code.